src/masternodeman.cpp

```diff
@@ -534,13 +534,15 @@ static void updateWinner(MasternodeRef& winner, int64_t& score, const uint256& h
 
 MasternodeRef CMasternodeMan::GetCurrentMasterNode(int nHeight, const uint256& hash) const
 {
+    int minProtocol = ActiveProtocol();
     int64_t score = 0;
     MasternodeRef winner = nullptr;
 
     // scan for winner
     for (const auto& it : mapMasternodes) {
-        if (!it.second->IsEnabled()) continue;
-        updateWinner(winner, score, hash, it.second);
+        const MasternodeRef& mn = it.second;
+        if (mn->protocolVersion < minProtocol || !mn->IsEnabled()) continue;
+        updateWinner(winner, score, hash, mn);
```

Comment:
same as in the other, would be good to squash ad2f71daa9b24564275ba632d72c5f3a1dabc6d0 into the commit that introduced the change 89416a1a1c2d9c7cfa97c139e7ac5555714923c5.

Plus, would be better to remove the `updateWinner` function and just write those three lines here. It makes the code harder to follow as it's a void static function updating the input argument that cannot be re-used in any other location.

Reply:
done